doc/user/content/ingest-data/_index.md

```diff
@@ -0,0 +1,244 @@
+description: "How to ingest data into Materialize from external systems."
+disable_list: true
+menu:
+  main:
+    identifier: "ingest-data"
+    name: "Ingest data"
+    weight: 11
+---
+
+Materialize can ingest data from various external systems:
+
+{{< multilinkbox >}}
+{{< linkbox title="Databases (CDC)" >}}
+- [PostgreSQL](/ingest-data/postgres/)
+- [MySQL](/ingest-data/mysql/)
+- [SQL Server](/ingest-data/cdc-sql-server/)
+- [CockroachDB](/ingest-data/cdc-cockroachdb/)
+{{</ linkbox >}}
+{{< linkbox title="Message Brokers" >}}
+- [Kafka](/ingest-data/kafka/)
+- [Redpanda](/sql/create-source/kafka)
+- [Other message brokers](/integrations/#message-brokers)
+{{</ linkbox >}}
+{{< linkbox title="Webhooks" >}}
+- [Amazon EventBridge](/ingest-data/webhooks/amazon-eventbridge/)
+- [Segment](/ingest-data/webhooks/segment/)
+- [Other webhooks](/sql/create-source/webhook)
+{{</ linkbox >}}
+{{</ multilinkbox >}}
+
+## Sources and clusters
+
+Materialize ingests data from external systems using
+[sources](/concepts/sources/). For the sources, you need to associate a
+[cluster](/concepts/clusters/) to provide the compute resources needed to ingest
+data.
+
+{{% tip %}}
+
+- If possible, dedicate a cluster just for sources.
+
+- When you create a new source, Materialize performs a one-time [snapshotting
+  operation](#snapshotting) to initially populate the source in Materialize.
+  Snapshotting is a resource-intensive operation that can require a significant
+  amount of CPU and memory. Consider using a larger cluster size during
+  snapshotting. Once the snapshotting operation is complete, you can downsize
+  the cluster to align with the steady-state ingestion.
+
+- See also [Best practices](#best-practices).
+
+{{% /tip %}}
+
+## Snapshotting
+
+Snapshotting refers to the initial population of a source in Materialize. When
+you create a new source, Materialize takes a snapshot of the data from the
+upstream (i.e., external) system at a given offset and loads it into
+Materialize.
+
+For the offset, Materialize chooses the latest available offset. For the
+available offset, Materialize uses:
+
+- Log Sequence Number (LSN) for PostgreSQL sources.
+
+- The number of transactions committed across all servers in the cluster for
+  MySQL sources.
+
+- Kafka message offset for Kafka sources.
+
+Materialize captures/commits a snapshot of all historical data up to that point.
+Snapshot is persisted in the storage layer, and all records in this initial load
+have the same ingestion timestamp.
+
+### Duration
+
+The snapshotting operation duration depends on the snapshot dataset size and the
+size of the Materialize cluster that is hosting the source. For very large
+sources that contain hundreds of GB of data, it can take up to an hour or even
+several hours to complete.
+
+{{% tip %}}
+
+- If possible, schedule creating new sources during off-peak hours to mitigate
+  the impact of snapshotting on both the upstream system and the Materialize
+  cluster.
```

Comment:
Need to verify about scheduling (namely, if there are exceptions that can be made (if necessary) to the rule depending on the source type)

Reply:
This is a weird one, and again only applies to production environments. Even then, it should be safe to _e.g._ add a new _subsource_ to a source during peak hours.